causalpy/experiments/synthetic_control.py

```diff
@@ -451,36 +492,83 @@ def get_plot_data_bayesian(self, hdi_prob: float = 0.94) -> pd.DataFrame:
+
+        # HDI intervals for predictions (always use treated_units dimension)
+        pre_hdi = get_hdi_to_df(
+            self.pre_pred["posterior_predictive"].mu.sel(treated_units=treated_unit),
+            hdi_prob=hdi_prob,
         )
-        pre_data[[pred_lower_col, pred_upper_col]] = get_hdi_to_df(
-            self.pre_pred["posterior_predictive"].mu, hdi_prob=hdi_prob
-        ).set_index(pre_data.index)
-        post_data[[pred_lower_col, pred_upper_col]] = get_hdi_to_df(
-            self.post_pred["posterior_predictive"].mu, hdi_prob=hdi_prob
-        ).set_index(post_data.index)
+        post_hdi = get_hdi_to_df(
+            self.post_pred["posterior_predictive"].mu.sel(treated_units=treated_unit),
+            hdi_prob=hdi_prob,
+        )
+
+        # Extract only the lower and upper columns and ensure proper indexing
+        pre_lower_upper = pre_hdi.iloc[:, [0, -1]].values  # Get first and last columns
+        post_lower_upper = post_hdi.iloc[:, [0, -1]].values
 
+        pre_data[[pred_lower_col, pred_upper_col]] = pre_lower_upper
+        post_data[[pred_lower_col, pred_upper_col]] = post_lower_upper
+
+        # Impact data - always use primary unit for main dataframe
         pre_data["impact"] = (
-            self.pre_impact.mean(dim=["chain", "draw"]).isel(treated_units=0).values
+            self.pre_impact.mean(dim=["chain", "draw"])
+            .sel(treated_units=treated_unit)
+            .values
         )
         post_data["impact"] = (
-            self.post_impact.mean(dim=["chain", "draw"]).isel(treated_units=0).values
+            self.post_impact.mean(dim=["chain", "draw"])
+            .sel(treated_units=treated_unit)
+            .values
+        )
+        # Impact HDI intervals (always use treated_units dimension)
+        pre_impact_hdi = get_hdi_to_df(
+            self.pre_impact.sel(treated_units=treated_unit), hdi_prob=hdi_prob
+        )
+        post_impact_hdi = get_hdi_to_df(
+            self.post_impact.sel(treated_units=treated_unit), hdi_prob=hdi_prob
         )
-        pre_data[[impact_lower_col, impact_upper_col]] = get_hdi_to_df(
-            self.pre_impact, hdi_prob=hdi_prob
-        ).set_index(pre_data.index)
-        post_data[[impact_lower_col, impact_upper_col]] = get_hdi_to_df(
-            self.post_impact, hdi_prob=hdi_prob
-        ).set_index(post_data.index)
+
+        # Extract only the lower and upper columns for impact HDI
+        pre_impact_lower_upper = pre_impact_hdi.iloc[:, [0, -1]].values
+        post_impact_lower_upper = post_impact_hdi.iloc[:, [0, -1]].values
+
+        pre_data[[impact_lower_col, impact_upper_col]] = pre_impact_lower_upper
+        post_data[[impact_lower_col, impact_upper_col]] = post_impact_lower_upper
 
         self.plot_data = pd.concat([pre_data, post_data])
 
         return self.plot_data
+
+    def _get_score_title(self, treated_unit: str, round_to=None):
```

Comment:
better type hints rount_to: Optional[fleat] = None and return str?

Reply:
Nice. Fixed in https://github.com/pymc-labs/CausalPy/pull/494/commits/3101cdc491622aa45f97013d597c9935e28f2f64 though I make it an Optional[int]